src/hotspot/share/interpreter/oopMapCache.cpp

```diff
@@ -276,26 +276,26 @@ bool OopMapCacheEntry::verify_mask(CellTypeState* vars, CellTypeState* stack, in
 
   // Check if map is generated correctly
   // (Use ?: operator to make sure all 'true' & 'false' are represented exactly the same so we can use == afterwards)
-  Log(interpreter, oopmap) logv;
-  LogStream st(logv.trace());
+  const bool log = log_is_enabled(Trace, interpreter, oopmap);
+  LogStream st(Log(interpreter, oopmap)::trace());
 
-  st.print("Locals (%d): ", max_locals);
+  if (log) st.print("Locals (%d): ", max_locals);
```

Comment:
This seems unfortunate to have these if statements, and a counter goal of the logging framework.  Is there another way to do this without the if statements?

Reply:
The problem is that here we use multiple print()/cr() statements to make up a single log message, thus we cannot simply use regular log_trace(). The only obvious way to avoid multiple if statements is to quote whole body with one if and introduce else branch without logging. But as I understand the intention of LogStream is exactly to combine a log message as a combination of pieces, so I don't see any problem here

Reply:
The logging framework was intended to avoid these explicit if statements also.  It seems it could be refactored to duplicate the loops with logging, and not combine them with verification.

Reply:
The logging framework was also supposed to not introduce overhead if logging is not enabled, but that is not the case as has been discovered here. I prefer the `if` statements to duplicating the code to have one block with logging and one without.

Reply:
fixed 34713cf